ui/src/app/shared/components/tri-state-button.component.ts

```diff
@@ -0,0 +1,86 @@
+import {Component,Input,Output,OnInit, AfterViewInit, DoCheck,
+  ViewChild,EventEmitter,ChangeDetectionStrategy,
+  ChangeDetectorRef, Renderer, ElementRef,
+  forwardRef} from '@angular/core';
+import {Pipe, PipeTransform} from '@angular/core';
+import {Observable, Subscription} from 'rxjs/Rx';
+import 'rxjs/add/operator/debounceTime';
+import 'rxjs/add/operator/throttleTime';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/distinctUntilChanged';
+
+import { Selectable } from '../../shared/model/selectable';
+
+@Component({
+  selector: 'scdf-tri-state-button',
+  template:
+  `<button #theButton name="topLevel" type="button"
+         class="btn btn-default"><span class="glyphicon glyphicon-trash"></span>
+    {{label}}
+  </button>`
+})
+export class TristateButton implements AfterViewInit, DoCheck, OnInit  {
+
+  public topLevel: boolean = false;
+  public _items: Array<Selectable> = [];
+
+  private _subscription: Subscription;
+  @Input()
+  items: Observable<any[]>;
+
+  label:String = '';
+
+  @ViewChild("theButton") button;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) { }
+
+  private setState() {
```

Comment:
All shared component should be declared in the `SharedModule`. Will address that.